apps/qubit/modules/informationobject/actions/updatePublicationStatusAction.class.php

```diff
@@ -35,19 +35,60 @@ public function execute($request)
             if ($this->form->isValid()) {
                 // Update resource synchronously
                 $publicationStatusId = $this->form->getValue('publicationStatus');
-                $this->resource->setPublicationStatus($publicationStatusId);
-                $this->resource->save();
 
-                // Update descendants using job scheduler
+                $sql = 'UPDATE status
+                    JOIN information_object io ON status.object_id = io.id
+                    SET status.status_id = :publicationStatus
+                    WHERE status.type_id = :publicationStatusType
+                    AND io.id = :id';
```

Comment:
No need to join here, just add another where clause for the `status.object_id`.

Reply:
I have removed this in favour of using $resource->save() with indexOnSave = false